scripts/core/superdesk-api.d.ts

```diff
@@ -226,8 +226,15 @@ declare module 'superdesk-api' {
         sign_off: string;
         feature_media?: any;
         media_description?: string;
-        associations?: any;
-        type: 'text' | 'picture' | 'video' | 'audio' | 'preformatted' | 'graphic' | 'composite';
+        associations?: IArticle | IRelatedArticle;
```

Comment:
`IRelatedArticle` needs to be exported. This is probably the only file that's not being typed checked properly. Can you also add a comment that `IRelatedArticle` is for internal items and `IArticle` for external like from search providers?

https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/apps/relations/directives/RelatedItemsDirective.ts#L222

Reply:
`IRelatedArticle` is already being exported in this file. About the comment, it's not like that, here's an example of associations:

![image](https://user-images.githubusercontent.com/4324982/71360912-a95ab080-2591-11ea-9b1f-c4fc6cef548f.png)

All of those are internal (i.e they are user created items) and yet they are `IRelatedArticle` and `IArticle`

Reply:
You are right about the `IRelatedArticle` being exported in this file.

I'm not sure about all associations, but related items should defintely be only stored in `IRelatedArticle` format. See `SDESK-3681`.

@petrjasek do you know what are all the cases when entire objects are stored? Items from custom search providers? What more? Custom media, galleries?

Reply:
custom media, galleries. we only use reference for related items, otherwise there is full object.